config/config.go

```diff
@@ -81,58 +81,30 @@ func (c Config) Validate() error {
 	if _, err := utility.StringToDuration(global.GetEvaluationInterval()); err != nil {
 		return fmt.Errorf("invalid rule evaluation interval: %s", err)
 	}
-	if err := c.validateLabels(global.Labels); err != nil {
+	if err := validateLabels(global.Labels); err != nil {
 		return fmt.Errorf("invalid global labels: %s", err)
 	}
 
-	// Check each job configuration for validity.
-	jobNames := map[string]bool{}
-	for _, job := range c.Job {
-		if jobNames[job.GetName()] {
-			return fmt.Errorf("found multiple jobs configured with the same name: '%s'", job.GetName())
-		}
-		jobNames[job.GetName()] = true
+	// Check each scrape configuration for validity.
+	jobNames := map[string]struct{}{}
+	for _, sc := range c.ScrapeConfigs() {
+		name := sc.GetJobName()
 
-		if !jobNameRE.MatchString(job.GetName()) {
-			return fmt.Errorf("invalid job name '%s'", job.GetName())
-		}
-		if _, err := utility.StringToDuration(job.GetScrapeInterval()); err != nil {
-			return fmt.Errorf("invalid scrape interval for job '%s': %s", job.GetName(), err)
-		}
-		if _, err := utility.StringToDuration(job.GetSdRefreshInterval()); err != nil {
-			return fmt.Errorf("invalid SD refresh interval for job '%s': %s", job.GetName(), err)
-		}
-		if _, err := utility.StringToDuration(job.GetScrapeTimeout()); err != nil {
-			return fmt.Errorf("invalid scrape timeout for job '%s': %s", job.GetName(), err)
-		}
-		for _, targetGroup := range job.TargetGroup {
-			if err := c.validateLabels(targetGroup.Labels); err != nil {
-				return fmt.Errorf("invalid labels for job '%s': %s", job.GetName(), err)
-			}
-			if err := c.validateHosts(targetGroup.Target); err != nil {
-				return fmt.Errorf("invalid targets for job '%s': %s", job.GetName(), err)
-			}
+		if _, ok := jobNames[name]; ok {
+			return fmt.Errorf("found multiple scrape configs configured with the same job name: %q", name)
```

Comment:
The current design requires identity for scrape configs - the default job name is the most reasonable way to enforce this.

Can you think of a common scenario where you'd want multiple scrape configs for the same job?
For advanced use cases relabelling will still allow circumventing this.